x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/30_ml_jobs_crud.yml

```diff
@@ -1,3 +1,8 @@
+setup:
+  - do:
+      indices.create:
+        index: .ml-state
```

Comment:
Please can you create this with realistic mappings and settings.  Something like:

```
  "settings" : {
    "index" : {
      "auto_expand_replicas" : "0-1"
    }
  },
  "mappings" : {
    "_doc": {
      "_meta": {
        "version": "7.0.0"
      },
      "enabled": false
    }
  }
```

Reply:
Aren't these settings going to be inherited from the template?
`x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/anomalydetection/state_index_template.json`
Let me verify that.

Reply:
Oh yes, good point

Reply:
> Aren't these settings going to be inherited from the template?
x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/anomalydetection/state_index_template.json
Let me verify that.

Confirmed, the settings are picked up from the template.
Now, I'm going to make these tests pass.